alpha-commons/src/test/java/at/ac/tuwien/kr/alpha/commons/terms/TermTest.java

```diff
@@ -90,4 +113,30 @@ public void testStringVsConstantSymbolEquality() {
 		// hashCode, i.e. return a non-zero result for non-equal objects
 		assertNotEquals(0, stringConstant.compareTo(constantSymbol));
 	}
+
+	@Test
+	public void testVariableSubstitution() {
+		Substitution substitution = new BasicSubstitution();
+		substitution.put(Terms.newVariable("X"), Terms.newConstant(1));
+
+		// Test variable term (trivial case, substituting results in constant).
+		VariableTerm var = VariableTermImpl.getInstance("X");
+		Term substitutedVar = var.substitute(substitution);
+		assertEquals(Terms.newConstant(1), substitutedVar);
```

Comment:
I think this should be `assertTrue( .. == .. )` as above.

Reply:
I think a check on `assertTrue( .. == .. )` has a fundamentally different meaning and should therefore be avoided in this case:
The test in question checks correct behavior of `substitute` for the various `Term` implementations. This is verified by checking whether the substitution result is equal to some expected term.
The check you suggest would completely rely on the fact that `Term` implementations are interned, i.e. that in our current implementations `t1.equals(t2)` iff `t1 == t2`. This is not publicly documented and not guaranteed to hold for future term implementations. It is, most importantly, a contract different (and independent) from the public contract of `substitute`and the two should be tested separately (as is the case, see `TermTest#testTermReferenceEquality`).